lib/wibox/widget/calendar.lua

```diff
@@ -117,6 +117,19 @@ local properties = { "date", "font", "spacing", "week_numbers", "start_sunday",
 -- @property fn_embed
 
 
+--- Return reproducible date for documentation if SOURCE_DATE_EPOCH is set
+local function normalize_date(date)
+    if not date then return nil end
+
+    local source_date_epoch = os.getenv("SOURCE_DATE_EPOCH")
+    if not source_date_epoch then return date end
+
+    local ret_date = os.date("!*t", source_date_epoch)
+    if date.month == nil then ret_date.month = nil end
+    if date.day == nil then ret_date.day = nil end
+    return ret_date
+end
```

Comment:
Do we really want this in `lib/`? It would also make awesome itself honor `SOURCE_DATE_EPOCH`.
Could you instead please try to get something like the following to work?
```diff
diff --git a/tests/examples/wibox/widget/defaults/calendar.lua b/tests/examples/wibox/widget/defaults/calendar.lua
index ea57daad9..39f9e9255 100644
--- a/tests/examples/wibox/widget/defaults/calendar.lua
+++ b/tests/examples/wibox/widget/defaults/calendar.lua
@@ -1,3 +1,13 @@
+local source_date_epoch = os.getenv("SOURCE_DATE_EPOCH") --DOC_HIDE
+if source_date_epoch then --DOC_HIDE
+    local gtable = require("gears.table") -- DOC_HIDE
+    local ret_date = os.date("!*t", source_date_epoch)
+    os.date = function(arg) --
+        assert(arg == "*t", "Expected '*t', but got: " .. arg) -- DOC_HIDE
+        return gtable.clone(ret_date) -- DOC_HIDE
+    end --
+end --DOC_HIDE
+
 local parent    = ... --DOC_HIDE
 local wibox     = require("wibox") --DOC_HIDE
 local beautiful = require( "beautiful" ) --DOC_HIDE
```
I guess since many tests need this, it would be best to have this code in some central file that is then `require()`d by the other files.

What do you think? Do you want to do this?

(Also: Thanks for showing how easy it is to use this variable in Lua! What's your context? Do you come from the debian reproducible build crowd? Since you are doing this work, it must be something like this, right?)